lib/internal/modules/cjs/loader.js

```diff
@@ -321,35 +324,24 @@ let patched = false;
 /* Make Module._cache case-insensitive on Windows */
 if (isWindows) {
   /* Create a proxy handler to intercept some operations */
+  const toLowerCaseIfString = (prop) => (typeof prop === 'string' ? StringPrototypeToLowerCase(prop) : prop);
```

Comment:
Is `StringPrototypeToLowerCase` the right way to convert the paths to be case-insensitive on Windows? `String.prototype.toLowerCase` follows the unicode case-folding rules, https://tc39.es/ecma262/multipage/text-processing.html#sec-string.prototype.tolowercase , I can't really find proper documentation about the case-folding rules of Windows so far (it seems FS-dependent), but my hunch tells me it's probably something that predates unicode so it's unlikely to be consistent with the output of `String.prototype.toLowerCase()`.

Reply:
From the Internet, at least https://archives.miloush.net/michkap/archive/2005/01/16/353873.html suggests that the path case folding rules on Windows differed from unicode for some early versions and didn't change much later - I doubt if they have switched to be aligned with unicode.